tests/Doctrine/Common/Lexer/CryptocurrencyAddressLexer.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\Common\Lexer;
+
+use Doctrine\Common\Lexer\AbstractLexer;
+
+use function array_values;
+use function count;
+
+class CryptocurrencyAddressLexer extends AbstractLexer
+{
+    /** @var string[] */
+    private $catchablePatterns = [
+        'bitcoin' => '^(bc1|[13])[a-zA-HJ-NP-Z0-9]{25,39}$',
+        'litecoin' => '^[LM3][a-km-zA-HJ-NP-Z1-9]{26,33}$',
+        'ethereum' => '^0x[a-fA-F0-9]{40}$',
+    ];
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function getCatchablePatterns()
+    {
+        return $this->catchablePatterns;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function getNonCatchablePatterns()
+    {
+        return [];
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @return string
+     */
+    protected function getType(&$value)
+    {
+        $items = $this->caughtByPatterns($value);
+
+        return array_values($items)[0] ?? '';
```

Comment:
:thinking: is returning an empty string really the appropriate thing to do here?